### PR TITLE
Build scripts for building NEXTSPACE from sources

### PR DESCRIPTION
**Pull request by onflapp**
Useful if you want to build everything from a source on your own.
This should also make it easier to build NEXTSPACE on distros other than RedHat or Debian.

### CODE REVIEW

**Comment by trunkmaster** [comment]
> 
> 
> Useful if you want to build everything from a source on your own.
> This should also make it easier to build NEXTSPACE on distros other than RedHat or Debian.

Are you sure? I see a lot of Debian-specifics (package names, patches, othe files).

**Comment by onflapp** [comment]
The scripts take the versions and patches from the Debian package definitions
This is to prevent having to maintain it twice.
Patches do not appear to be Debian distribution dependent.
Other than that, it is just cloning the repo, checking out specific branch, applying patches and building the source code.

@trunkmaster  do you want me to maintain versions and patches separately from the Debian package definitions?

**Comment by trunkmaster** [comment]
@onflapp you stated "...to build NEXTSPACE on distros other than RedHat or Debian.". And I see Debian specifics instead. Let me reframe my question: "What distros this scripts aimed for? What distros did you test?".

**Comment by onflapp** [comment]
Good point.

**Comment by onflapp** [comment]
@trunkmaster I tested it on FreeBSD.
Source preparation scripts all run fine. 
Not everything will compile obviously (yet) but that's not point of these scripts.
These scripts should just make it easier to prepare everything in one place without any dependencies to any particular distro and kick off the build process.

File: Packaging/Sources/install-dependencies-debian.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+if [ -z "$1" ];then
+	echo "specify list of dependencies: "
+	ls -1 *dependencies-debian*.txt
+	exit 1
+fi
+
+for DD in `cat $1` ;do
+  echo $DD
+  apt-get install -y $DD || exit 1
+done
```
[the remark this answers is not in the document]
Comment (onflapp) [reply Packaging/Sources/install-dependencies-debian.sh]: These are optional scripts that can help users install dependencies on minimal system.
I expect to add non-debian script in the future.

I moved these scripts to "extras"

File: Packaging/Sources/dependencies-debian9.txt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+libbsd-dev
+libudisks2-dev
+libdbus-1-dev
+libxkbfile-dev
+libpulse-dev
+libxmu-dev
+libxft-dev
+libxpm-dev
+libssl-dev
+libart-2.0-dev
```
[the remark this answers is not in the document]
Comment (onflapp) [reply Packaging/Sources/dependencies-debian9.txt]: I moved it scripts to "extras"

File: Packaging/Sources/9_install_scripts.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+if [ -d "$HOME/Library" ];then
+	echo "$HOME/Library exists already, please make sure it is up to date"
+else
+	cp -R /etc/skel/Library  $HOME/
+fi
+if [ -f "$HOME/.xinitrc" ];then
+	echo "$HOME/.xinitrc exists already, will not overwrite"
+else
+	cp ./copy_to_users_home/.xinitrc $HOME/
+fi
```
[the remark this answers is not in the document]
Comment (onflapp) [reply Packaging/Sources/9_install_scripts.sh]: This is minimal install to get NEXTSPACE working.
xinitrc seems to be working consistently across multiple distributions and platforms.
If user wants to get Login.app working, they may have to do system specific configuration.

**Comment by onflapp** [comment]
@trunkmaster I moved all "debian" related stuff out of the way now.
I thought patches need to applied first, but you are right, it will build without any patching.

The idea is to have very simple build script(s).

1. fetches sources
2. builds the stuff
3. make install

Building NEXTSPACE on distros other than RedHat or Debian is not that straight forward right now. 
As the existing build scripts build *packages* and then install them, you can easily run into package versioning problems etc, etc.

This build script will build NEXTSPACE directly from source code and `make install` it. There is no other magic there.
Furthermore, it is not intended to replace distro specific scripts, the intention is to get you started, even if you don't have one of supported distros or you just want to "roll your own".

File: Packaging/Sources/5_build_libs-gui.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+. /Developer/Makefiles/GNUstep.sh
+
+cd ./nextspace-gui-0.28.0+nextspace || exit 1
+
+make clean
+sh ./configure \
+	--disable-icu-config
```
Comment (trunkmaster) [inline Packaging/Sources/5_build_libs-gui.sh:9]: Why ICU library is disabled here? Do you have problems with it?

File: Packaging/Sources/6_build_libs-back.sh [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+. /Developer/Makefiles/GNUstep.sh
+
+cd ./nextspace-back-0.28.0+nextspace || exit 1
+
+make clean
+./configure \
+  --enable-graphics=art \
+  --with-name=art
+
+make install
```
Comment (trunkmaster) [inline Packaging/Sources/6_build_libs-back.sh:12]: `make fonts=no install`

File: Packaging/Sources/5_build_libs-gui.sh (hunk 4 above)
Comment (onflapp) [reply Packaging/Sources/5_build_libs-gui.sh:9]: hmm, I just copied this from the Debian.
OK, I'll double check!

**Comment by onflapp** [comment]
@trunkmaster I think I am done now.
many thanks!

**Pull request merged**